server/webapp/WEB-INF/rails.new/app/assets/javascripts/pipeline_configs/models/argument.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define(['mithril', 'lodash'], function (m, _) {
+
+  var Argument = function(data) {
+    this.data = m.prop(data);
+
+    this.isList = function () {
+      return _.isArray(this.data());
+    };
+
+    this.toJSON = function () {
+      if(_.isEmpty(this.data())) {
+        return {};
+      }
+      return this.isList() ? {arguments: this.data()} : {args: this.data()};
+    }
+  };
+
+  Argument.create = function(args, arguments) {
+    return arguments ? new Argument(arguments) : new Argument(args || '');
```

Comment:
`arguments` could be ambiguous because of [this](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/arguments)